### PR TITLE
Port the WorldAnchorManager from the HTK to MRTKv2.

### PR DESCRIPTION
**Pull request by wiwei**
https://github.com/microsoft/MixedRealityToolkit-Unity/issues/4509

The WorldAnchorManager provided a convenience wrapper on top of Unity's WorldAnchor/AnchorStore abstractions. This change ports the file mostly as-is into the Experimental namespace, because we haven't yet figured out how best to have end to end tests for this, and because we don't have any documentation yet.

Note that there have been several changes here from the HTK version:

1) Some functions that were protected/virtual have been made private, for the sake of going out with the minimal API surface needed to use the World Anchor Manager. This doesn't mean that we they will be private forever, just that we don't want to create a massive surface and want to extend it as-needed.
2) Removed usage of the Singleton<> template - it's up to the consumer how they choose to use this (they can do their own singleton if they want)

## Changes
- Fixes: #4509

### CODE REVIEW

File: Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/WorldAnchorManager.cs [hunk 1]
```diff
@@ -0,0 +1,604 @@
...
+using UnityEngine.XR.WSA;
+using UnityEngine.XR.WSA.Persistence;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Utilities
+{
+    /// <summary>
+    /// Wrapper around world anchor store to streamline some of the persistence API busy work.
+    /// </summary>
+    public class WorldAnchorManager : MonoBehaviour
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/WorldAnchorManager.cs]: Re: removal of Singleton

Was there a reason this wasn't adapted into something like an extension service?

Comment (wiwei) [reply Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/WorldAnchorManager.cs]: My goal here was to get a straight, minimal port in to unblock folks who needed it for historical reasons.

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/WorldAnchorManager.cs]: Looking at the code (not just the comments :) ), this is in the experimental namespace and should be safe to deliver initially as a monobehaviour to get developer feedback.

File: Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/WorldAnchorManager.cs [hunk 2]
```diff
@@ -0,0 +1,608 @@
...
+using System;
+using System.Collections.Generic;
+using UnityEngine.XR.WSA;
+using UnityEngine.XR.WSA.Persistence;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Utilities
+{
+    /// <summary>
+    /// Wrapper around world anchor store to streamline some of the persistence API busy work.
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/WorldAnchorManager.cs]: nit: _maybe_ a bit too casual of a tone in the doc comments?

File: Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/WorldAnchorManager.cs [hunk 3]
```diff
@@ -0,0 +1,608 @@
...
+    /// </summary>
+    /// <remarks>
+    /// This class only functions when built for the WSA platform. It uses APIs that are only present
+    /// on that platform.
+    /// </remarks>
+    public class WorldAnchorManager : MonoBehaviour
+    {
+        /// <summary>
+        /// Debug text for displaying information.
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/WorldAnchorManager.cs]: ?: What kind of information?

File: Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/WorldAnchorManager.cs [hunk 4]
```diff
@@ -0,0 +1,608 @@
...
+        /// </summary>
+        private struct AnchorAttachmentInfo
+        {
+            public GameObject AnchoredGameObject { get; set; }
+            public string AnchorName { get; set; }
+            public AnchorOperation Operation { get; set; }
+        }
+
+        /// <summary>
+        /// The data structure for anchor operations.
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/WorldAnchorManager.cs]:
```suggestion
        /// Enumeration defining the types of anchor operations.
```

File: Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/WorldAnchorManager.cs [hunk 5]
```diff
@@ -0,0 +1,608 @@
...
+            }
+
+            var anchors = FindObjectsOfType<WorldAnchor>();
+
+            if (anchors == null)
+            {
+                return;
+            }
+
+            for (var i = 0; i < anchors.Length; i++)
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/WorldAnchorManager.cs]: nit: why var when int is just as few characters? :)

Comment (wiwei) [reply Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/WorldAnchorManager.cs]: Old code, changed.

File: Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/WorldAnchorManager.cs [hunk 6]
```diff
@@ -0,0 +1,616 @@
...
```
Comment (wiwei) [inline Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/WorldAnchorManager.cs]:
```suggestion
            for (int i = 0; i < anchors.Length; i++)
```

**Pull request merged**